src/test/enhetstester/kotlin/no/nav/familie/ks/sak/kjerne/personopplysninggrunnlag/PersonopplysningGrunnlagServiceTest.kt

```diff
@@ -191,6 +199,88 @@ internal class PersonopplysningGrunnlagServiceTest {
+        )
+    }
+
+    @Test
+    fun `oppdaterPersonopplysningGrunnlag - skal deaktivere eksisterende aktivt personopplysningsgrunnlag og opprette et nytt med barn fra søknad og vedtatt behandling i revurdering`() {
+        val lagredePersonopplysningsGrunnlag = mutableListOf<PersonopplysningGrunnlag>()
+        val søker = randomAktør()
+        val barn1 = randomAktør()
+        val barn2 = randomAktør()
+        val barn3 = randomAktør()
+        val fagsak = lagFagsak(søker)
+        val vedtattBehandling = lagBehandling(fagsak, opprettetÅrsak = BehandlingÅrsak.SØKNAD)
+        val behandling = lagBehandling(fagsak, opprettetÅrsak = BehandlingÅrsak.SØKNAD)
+        val eksisterendePersonopplysningGrunnlag = lagPersonopplysningGrunnlag(
+            behandlingId = vedtattBehandling.id,
+            søkerPersonIdent = søker.aktivFødselsnummer(),
+            barnasIdenter = listOf(barn1.aktivFødselsnummer())
+        )
+
+        val nyttPersonopplysningGrunnlag = lagPersonopplysningGrunnlag(
+            behandlingId = behandling.id,
+            søkerPersonIdent = søker.aktivFødselsnummer()
+        ).also { it.personer.clear() }
+
+        val nyttPersonopplysningGrunnlagMedBarna = lagPersonopplysningGrunnlag(
+            behandlingId = behandling.id,
+            søkerPersonIdent = søker.aktivFødselsnummer(),
+            barnasIdenter = listOf(barn1.aktivFødselsnummer(), barn2.aktivFødselsnummer(), barn3.aktivFødselsnummer())
+        )
+
+        every { personopplysningGrunnlagRepository.findByBehandlingAndAktiv(any()) } returns eksisterendePersonopplysningGrunnlag
+        every {
+            andelTilkjentYtelseRepository.finnAndelerTilkjentYtelseForBehandlingOgBarn(
+                any(),
+                any()
+            )
+        } returns listOf(
+            mockk()
+        )
+        every { personidentService.hentOgLagreAktør(any(), any()) } returns barn2 andThen barn3
+        every { personService.lagPerson(any(), any(), any(), any(), any()) } returns lagPerson(
+            personopplysningGrunnlag = nyttPersonopplysningGrunnlag,
+            aktør = søker
+        ) andThen lagPerson(
+            personopplysningGrunnlag = nyttPersonopplysningGrunnlag,
+            aktør = barn1,
+            personType = PersonType.BARN
+        ) andThen lagPerson(
+            personopplysningGrunnlag = nyttPersonopplysningGrunnlag,
+            aktør = barn2,
+            personType = PersonType.BARN
+        ) andThen lagPerson(
+            personopplysningGrunnlag = nyttPersonopplysningGrunnlag,
+            aktør = barn3,
+            personType = PersonType.BARN
+        )
+        every { personopplysningGrunnlagRepository.saveAndFlush(capture(lagredePersonopplysningsGrunnlag)) } returns mockk()
+        every { personopplysningGrunnlagRepository.save(capture(lagredePersonopplysningsGrunnlag)) } returns nyttPersonopplysningGrunnlag andThen nyttPersonopplysningGrunnlagMedBarna
```

Comment:
Kan bruke returnsArgument
```suggestion
        every { personopplysningGrunnlagRepository.save(any()) } returnsArgument 0
```

https://mockk.io/
<img width="858" alt="image" src="https://user-images.githubusercontent.com/17828446/199716144-a5628661-f33a-4bcb-a155-bf6b58f3c589.png">